### PR TITLE
Add shim for EMR 3.0.1 and EMR 3.0.1-SNAPSHOT

### PR DESCRIPTION
**Pull request by tgravescs**
This adds a new shim layer for EMR version 3.0.1 and 3.0.1-snapshot.  They told us there are no differences in api between it an apache 3.0.1. The actually made the file scan RDD that was in compatible in 3.0.0 emr now compatible, so I removed that change as well.  So this shim is based on the Spark301Shims one and is just a version number check.

I tested this by building apache spark with the version 3.0.1-amzn-0 and 3.0.1-amzn-0-SNAPSHOT and ran a few basic queries. 
20/10/21 14:13:46 INFO ShimLoader: Loading shim for Spark version: 3.0.1-amzn-0

I have not run integration tests on this since emr 3.0.1 is not available yet but we will work with them to do so. They specifically requested support for the -SNAPSHOT for testing.

I also noticed we still had snapshot support for 3.0.1 so I removed that. If folks want I can split that apart.

fixes https://github.com/NVIDIA/spark-rapids/issues/905

### CODE REVIEW

**Comment by tgravescs** [comment]
build

File: shims/spark301emr/src/main/scala/com/nvidia/spark/rapids/shims/spark301emr/Spark301EMRShims.scala [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import com.nvidia.spark.rapids.spark301emr.RapidsShuffleManager
+
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.execution.datasources.{FilePartition, FileScanRDD, PartitionedFile}
+
+class Spark301EMRShims extends Spark301Shims {
+
+  private var fileScanRddConstructor: Option[Constructor[_]] = None
```
Comment (tgravescs) [inline shims/spark301emr/src/main/scala/com/nvidia/spark/rapids/shims/spark301emr/Spark301EMRShims.scala:32]: oops need to remove this

Comment (tgravescs) [reply shims/spark301emr/src/main/scala/com/nvidia/spark/rapids/shims/spark301emr/Spark301EMRShims.scala:32]: and cleanup imports

**Comment by tgravescs** [comment]
build

**Review by jlowe — APPROVED** [review]

**Pull request merged**